### PR TITLE
Upgrade go to v1.17

### PR DESCRIPTION
**Pull request by bdunne**

### CODE REVIEW

**Comment by bdunne** [comment]
I was running into the following errors because I was using the newer binary, but `GOROOT` was pointing to the old version.
```
$ operator-sdk run local
INFO[0000] Running the operator locally; watching namespace "33" 
# internal/race
compile: version "go1.14.10" does not match go tool version "go1.17.5"
# unicode
compile: version "go1.14.10" does not match go tool version "go1.17.5"
# math/bits
compile: version "go1.14.10" does not match go tool version "go1.17.5"
# runtime/internal/sys
compile: version "go1.14.10" does not match go tool version "go1.17.5"
# unicode/utf8
compile: version "go1.14.10" does not match go tool version "go1.17.5"
# unicode/utf16
compile: version "go1.14.10" does not match go tool version "go1.17.5"
# k8s.io/apimachinery/pkg/selection
compile: version "go1.14.10" does not match go tool version "go1.17.5"
# encoding
compile: version "go1.14.10" does not match go tool version "go1.17.5"
# internal/cpu
flag provided but not defined: -p
usage: asm [options] file.s ...
Flags:
  -D value
    	predefined symbol with optional simple value -D=identifier=value; can be set multiple times
  -I value
    	include directory; can be set multiple times
  -S	print assembly and machine code
  -V	print version and exit
  -debug
    	dump instructions as they are parsed
  -dynlink
    	support references to Go symbols defined in other shared libraries
  -e	no limit on number of errors reported
  -gensymabis
    	write symbol ABI information to output file, don't assemble
  -newobj
    	use new object file format
  -o string
    	output file; default foo.o for /a/b/c/foo.s as first argument
  -shared
    	generate code that can be linked into a shared library
  -trimpath string
    	remove prefix from recorded source file paths
# runtime/internal/atomic
...
```

**Comment by Fryguy** [comment]
Seems the zz_generated file needs to be go formatted?

File: manageiq-operator/pkg/apis/manageiq/v1alpha1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
```
Comment (Fryguy) [inline manageiq-operator/pkg/apis/manageiq/v1alpha1/zz_generated.deepcopy.go:1]: LOL - did you add this or did something do this for you?

Comment (bdunne) [reply manageiq-operator/pkg/apis/manageiq/v1alpha1/zz_generated.deepcopy.go:1]: That was added by `gofmt -w .`

Comment (bdunne) [reply manageiq-operator/pkg/apis/manageiq/v1alpha1/zz_generated.deepcopy.go:1]: Oh good, this conflicts with operator-sdk generated code...

**Comment by Fryguy** [comment]
Backported to `najdorf` in commit 5ddaed9a26a3e5c0e2e75e6e9b5713bfa04296fb.

```text
commit 5ddaed9a26a3e5c0e2e75e6e9b5713bfa04296fb
Author: Jason Frey <fryguy9@gmail.com>
Date:   Mon Feb 21 15:01:44 2022 -0500

    Merge pull request #816 from bdunne/upgrade_go
    
    Upgrade go to v1.17
    
    (cherry picked from commit 98c2220a425f0b9f6673f899eb44348f4df126f0)
```

**Pull request merged**